### DISCUSSION TITLE
Analysis based on group.by gRNAs

### DISCUSSION THREAD

**Discussion by nhu-github** [Katsevich-Lab/sceptre#121] [Q&A] [answered]
Hi,
Is there a way to group.by gRNAs to do analysis? For example, one gene has 4 gRNAs designed in experiment, we would like to agrregate those four counts into one.  One option would be in the beginning, we do aggregate before import data to sceptre, but I am not sure if this practice would impact gRNA assignment results. Another situation is after one round analysis on induviual gRNA analysis, can we use the first round data to perform aggregation of gRNAs within sceptre?

**Comment by timothy-barry** [comment]
Hi Nan,

Have you had a chance to look at the [gRNA integration strategy](https://timothy-barry.github.io/sceptre-book/set-analysis-parameters.html#sec-grna_integration_strategy) portion of the book? There are three gRNA integration strategies within the software: `singleton`, `bonferroni`, and `union`.

**Reply by nhu-github** [reply]
Yes, I have run one cis analysis on `singleton` and have an sceptre object with gRNA assigned with `singleton` strategy. Now I am going to do cis perturb inference but with `union` gRNA setting. I am wondering if I can use the `singleton` assigned gRNA  sceptre object for this analysis, and somewhere there is a parameter could `union` the assigned gRNA matrix for me? If not, I will need to redo set-analysis-parameter step and redo the gRNA assignment with `union`.

**Reply by timothy-barry** [reply, marked as answer]
Ah, I see, so another question about caching/optimization :)

Unfortunately, no, this is not currently possible. You will have to rerun `assign_grnas()` if you want to change the grouping strategy from `singleton` to `union`. We hope to add more sophisticated caching of the gRNA assignments in the future.

Just out of curiosity, are you using the mixture gRNA assignment strategy? As you explore different analysis parameters, you may want to use the simpler and faster thresholding or maximum assignment strategy.

**Reply by nhu-github** [reply]
Yes, currently only using `mixture` gRNA assignment strategy  as my data is high-MOI. So first follow a default setting. Probably `threading` is much faster than `mixture`. But do you think `threading=5` will give me similar results as `mixture`?

**Reply by timothy-barry** [reply]
Yes, `thresholding` is much faster than `mixture`. If your gRNA count distributions exhibit good separation, then to first approximation, I would expect thresholding to produce roughly similar answers to mixture. Therefore, if you want to play around with the parameters in `set_analysis_parameters()`, you might consider using the thresholding method as a starting point. Then, when you have identified a set of analysis parameters that you like, you could use the mixture method to (hopefully) obtain slightly more precise results.
